tests/api-tests/test-runner.ts

```diff
@@ -1,183 +1,203 @@
+async function getTestPrismaModule (prismaSchemaPath: string, schema: string) {
+  if (prismaModuleCache.has(schema)) return prismaModuleCache.get(schema)!
+  return prismaModuleCache.set(schema, await getTestPrismaModuleInner(prismaSchemaPath, schema)).get(schema)!
+}
+
+const deferred: (() => Promise<void>)[] = []
 afterAll(async () => {
-  await fs.rm(prismaSchemaDirectory, { recursive: true, force: true })
+  for (const f of deferred) {
+    await f()
+  }
 })
 
-let hasCreatedDatabase = false
-
-async function pushSchemaToDatabase (schema: string) {
-  if (dbProvider === 'sqlite') {
-    // touch the file (or truncate it), easiest way to start from scratch
-    await fs.writeFile(path.join(prismaSchemaDirectory, SQLITE_DATABASE_FILENAME), '')
-    await withMigrate(prismaSchemaPath, migrate =>
-      runMigrateWithDbUrl(dbUrl, undefined, () =>
-        migrate.engine.schemaPush({
-          force: true,
-          schema,
-        })
-      )
-    )
-    return
-  }
-  let justCreatedDatabase = hasCreatedDatabase
-    ? false
-    : await createDatabase(dbUrl, prismaSchemaDirectory)
-  await withMigrate(prismaSchemaPath, async migrate => {
-    if (!justCreatedDatabase) {
-      await runMigrateWithDbUrl(dbUrl, undefined, () => migrate.reset())
+async function testConfig2<TypeInfo extends BaseKeystoneTypeInfo> (config: KeystoneConfig<TypeInfo>) {
+  const tmp = join(tmpdir(), `ks6-tests-${randomBytes(8).toString('base64url')}`)
+  await fs.mkdir(tmp)
+  deferred.push(async () => {
+    await fs.rm(tmp, { recursive: true, force: true })
+  })
+
+  return initConfig({
+    types: {
+      path: join(tmp, 'test-types.ts')
+    },
+    ...config,
+    db: {
+      ...config.db,
+      // see api-tests/utils.ts
+      url: config.db.url === 'file:./test.db' ? `file:${join(tmp, 'test.db')}` : config.db.url,
+      prismaClientPath: join(tmp, '.client'),
+      prismaSchemaPath: join(tmp, 'schema.prisma'),
+    },
+    graphql: {
+      schemaPath: join(tmp, 'schema.graphql'),
+      ...config.graphql,
     }
-    await runMigrateWithDbUrl(dbUrl, undefined, () =>
-      migrate.engine.schemaPush({
-        force: true,
-        schema,
-      })
-    )
   })
-  hasCreatedDatabase = true
 }
 
-let lastWrittenSchema = ''
+export async function setupTestEnv<TypeInfo extends BaseKeystoneTypeInfo> (
+  config_: KeystoneConfig<TypeInfo>
+) {
+  const config = await testConfig2(config_)
+  const { graphQLSchema, getKeystone } = createSystem(config)
+  const artifacts = await generatePrismaAndGraphQLSchemas('', config, graphQLSchema)
 
-export async function setupTestEnv<TypeInfo extends BaseKeystoneTypeInfo> ({
-  config: _config,
-}: {
-  config: KeystoneConfig<TypeInfo>
-}): Promise<TestEnv<TypeInfo>> {
-  // Force the UI to always be disabled.
-  const config = initConfig({
-    ..._config,
-    ui: { ..._config.ui, isDisabled: true },
-  })
+  // setup (reset) the database
+  await withMigrate(config.db.prismaSchemaPath, async migrate => {
```

Comment:
Looks like by doing this instead of `pushSchemaToDatabase` the database is not being created you need to run 
```ts
await createDatabase(config.db.url, path.dirname(config.db.prismaSchemaPath))
```
before the migration (`createDatebase` is from `@prisma/internals`

Reply:
Yes! I have fixed this locally, but there a few other subleties that came up so I might do this across two pull requests